### ISSUE TITLE
Fix pyflakes for mail

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6999]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#6999|
|Type|enhancement|
|Created|2014-02-27 13:14:36Z|
|Branch|https://github.com/twisted/twisted/tree/pyflakes-mail-6999|

Attachments:

* [6999-1.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cac/cac6f9be4458fc3b0106f31eaeee90c7c67209ef/b29ae8959eb4bde745b7ea618a9f50bf5f1a7539.diff) (9928 bytes) - added by adiroiban on 2014-02-27 13:39:23Z - 
* [6999-2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cac/cac6f9be4458fc3b0106f31eaeee90c7c67209ef/6116f70bb4c265f889f5eacd165cc8d6ac748b5e.diff) (13380 bytes) - added by adiroiban on 2014-04-14 08:13:57Z - 
* [6999-3.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/cac/cac6f9be4458fc3b0106f31eaeee90c7c67209ef/79cf85b46ed2455f4bd540126a72d07141210137.diff) (726 bytes) - added by adiroiban on 2014-04-14 13:07:12Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6999 6999
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__ 
branch__branches_pyflakes_mail_6999 branches/pyflakes-mail-6999
branch_author__rwall rwall
status__closed closed
resolution__fixed fixed
component__mail mail
keywords__ 
time__1393506876000000 1393506876000000
changetime__1397659546000000 1397659546000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban commented|
|-|-|

In bounce.py, vars() was doing pretty magic/implicit stuff. I have updated to code to explicit string formating. That might be cute/smart but I think that is bad code.

Other changes are just removal of unused imports or variables

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)|@wallrj set owner to @wallrj<br>@wallrj set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_3"></a>|@wallrj set owner to @adiroiban<br>@wallrj set status to `new`|
|-|-|

Thanks Adi,

These changes look pretty good, but there are some test failures and problems
which I have noted below.

Notes:
 * No pyflakes warnings after applying the patch.

Points:
 1. [twisted/mail/protocols.py](twisted/mail/protocols.py)
    1. The "imports for pydoctor" seem pretty bad. Are you sure they're
       necessary? I thought pydoctor was able to resolve absolute twisted module
       paths. If not then raise a pydoctor ticket and we'll just live with some
       mising API links until pydoctor gets fixed. Link to the pydoctor ticket
       from this ticket and vice versa.
    2. And the "twisted to silence the linter" is also bad but won't be
       necessary if you remove the two imports above, I guess.
 2. [twisted/mail/test/test_options.py](twisted/mail/test/test_options.py)
    1. "OpenSSL # Silence the linter." I think you said you intended to replace
       these with t.p.reflect.requireModule.
 3. Test Failures
    1. trial reports test failures after applying your patch (see below)

Please answer or address the numbered points above and submit a new patch
against trunk.

## Trial Test Failures
```
[richard@zorin trunk]$ trial twisted.mail
twisted.mail
  test
    test_bounce ...                                                     [ERROR]
    test_imap ...                                                       [ERROR]
    test_mail ...                                                       [ERROR]
    test_mailmail ...                                                   [ERROR]
    test_options ...                                                    [ERROR]
    test_pop3 ...                                                       [ERROR]
    test_pop3client ...                                                 [ERROR]
twisted.mail.test.test_scripts
  ScriptTests
    test_mailmail ...                                                    [FAIL]
twisted.mail
  test
    test_smtp ...                                                       [ERROR]

===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_scripts.py", line 18, in test_mailmail
    self.scriptTest("mail/mailmail")
  File "/home/richard/projects/Twisted/trunk/twisted/scripts/test/test_scripts.py", line 81, in scriptTest
    self.assertIn(str(version), scriptVersion)
  File "/home/richard/projects/Twisted/trunk/twisted/trial/_synctest.py", line 416, in assertIn
    % (containee, container))
twisted.trial.unittest.FailTest: '13.2.0' not in ''

twisted.mail.test.test_scripts.ScriptTests.test_mailmail
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_bounce.py", line 9, in <module>
    from twisted.mail import bounce
  File "/home/richard/projects/Twisted/trunk/twisted/mail/bounce.py", line 17, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_bounce
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_imap.py", line 22, in <module>
    from twisted.mail.imap4 import MessageSet
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_imap
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_mail.py", line 22, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_mail
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_mailmail.py", line 13, in <module>
    from twisted.mail.scripts.mailmail import parseOptions
  File "/home/richard/projects/Twisted/trunk/twisted/mail/scripts/mailmail.py", line 22, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_mailmail
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_options.py", line 11, in <module>
    from twisted.mail import protocols
  File "/home/richard/projects/Twisted/trunk/twisted/mail/protocols.py", line 10, in <module>
    from twisted.mail import pop3
  File "/home/richard/projects/Twisted/trunk/twisted/mail/pop3.py", line 21, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_options
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_pop3.py", line 19, in <module>
    import twisted.mail.protocols
  File "/home/richard/projects/Twisted/trunk/twisted/mail/protocols.py", line 10, in <module>
    from twisted.mail import pop3
  File "/home/richard/projects/Twisted/trunk/twisted/mail/pop3.py", line 21, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_pop3
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_pop3client.py", line 10, in <module>
    from twisted.mail.pop3 import AdvancedPOP3Client as POP3Client
  File "/home/richard/projects/Twisted/trunk/twisted/mail/pop3.py", line 21, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_pop3client
===============================================================================
[ERROR]
Traceback (most recent call last):
  File "/home/richard/projects/Twisted/trunk/twisted/trial/runner.py", line 498, in loadPackage
    module = modinfo.load()
  File "/home/richard/projects/Twisted/trunk/twisted/python/modules.py", line 383, in load
    return self.pathEntry.pythonPath.moduleLoader(self.name)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 303, in namedAny
    topLevelPackage = _importAndCheckStack(trialname)
  File "/home/richard/projects/Twisted/trunk/twisted/python/reflect.py", line 242, in _importAndCheckStack
    return __import__(importName)
  File "/home/richard/projects/Twisted/trunk/twisted/mail/test/test_smtp.py", line 14, in <module>
    from twisted.mail import smtp
  File "/home/richard/projects/Twisted/trunk/twisted/mail/smtp.py", line 1833, in <module>
    from twisted.mail.imap4 import IClientAuthentication
  File "/home/richard/projects/Twisted/trunk/twisted/mail/imap4.py", line 384, in <module>
    class LOGINCredentials(cred.credentials.UsernamePassword):
exceptions.AttributeError: 'module' object has no attribute 'credentials'

twisted.mail.test.test_smtp
-------------------------------------------------------------------------------
Ran 9 tests in 2.290s
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban removed owner|
|-|-|

Hi,

I have attached a new patch. All test pass. pyflakes pass.

I used the namedAny to work around pyflakes.

I have removed imports for pydoctor as they are not needed.

Sorry for uploading the patch before running the tests.

In imap.py the imports were building the 'twisted.cred' namespace on the fly. I think that this is obscure feature of import and I went for explicit imports.

Thanks

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_5"></a>|@wallrj commented|
|-|-|

(In [42202]) Branching to 'pyflakes-mail-6999'

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_6"></a>|@wallrj commented|
|-|-|

(In [42203]) Apply 6999-2.diff from adiroban. Refs [#6999](https://github.com/twisted/twisted/issues/6999)

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_7"></a>|@wallrj set owner to @adiroiban|
|-|-|

Thanks Adi,

Notes:
 * No pyflakes warnings
 * All twisted.mail tests still pass locally
 * Started build

Points:
 1. [branches/pyflakes-mail-6999/twisted/mail/test/test_options.py](branches/pyflakes-mail-6999/twisted/mail/test/test_options.py)
    1. Build failures: https://buildbot.twistedmatrix.com/builders/py-without-modules/builds/2352/steps/trial/logs/problems
    2. I personally think it would be clearer to use the new requireModule function here eg
       ```
       if requireModule('OpenSSL', default=None) is None:
           sslSkip = 'OpenSSL could not be imported'
       else:
           sslSkip = None
       ```
 2. When you say the "building the 'twisted.cred' namespace on the fly." what exactly do you mean? I hadn't heard of this feature.

Please answer or address the numbered points above and submit a patch against log:branches/pyflakes-mail-6999

-RichardW.

## Conversation with adiroban in #twisted-dev
```
13:04 < rwall> adiroiban1: Hi. How come you didn't use your new "requireModule" function in #6999?
13:04 < rwall> https://twistedmatrix.com/trac/ticket/7014
13:04 < adiroiban1> :) ...requireModule is not mine... that is collective code  :)
13:05 < adiroiban1> let me see
13:05 < rwall> adiroiban1: But you wrote it, right? :)
13:05 < adiroiban1> yes... but after sending all those patches
13:05 < rwall> ok.
13:06 < rwall> Looks like it should be used in #6999 anyway.
13:06 < adiroiban1> requireModule was as in a review for one of that patch
13:06 < rwall> yep. But it's been merged now so we can use it.
13:06 < adiroiban1> in 6999 i used namedAny
13:07 < adiroiban1> requireModule was designed for another use case
13:07 < adiroiban1> it was designed for production code... and not testing code to fill skip variable
13:08 < rwall> Ah, I see. The import error gets assigned to the skip message.
13:08 < adiroiban1> the idea is that requireModule will return a module
13:08 < adiroiban1> skip should be None when module exists
13:09 < adiroiban1> so we need a new helper for tests
13:09 < adiroiban1> skip = skipRequire('some.module') ... or something like this
13:09 < adiroiban1> which returns None if module exists
13:09 < adiroiban1> and a string when it does not exist
13:09 < rwall> if requireModule('OpenSSL') is None: sslSkip = 'OpenSSL not available'
13:10 < adiroiban1> something like this...
13:10 < adiroiban1> if requireModule('OpenSSL', **default=None**) is None: sslSkip = 'OpenSSL not available'
13:11 < adiroiban1> as otherwiser requireModule will raise ImportError
13:12 -!- fijal_ [~fijal@...] has joined #twisted-dev
13:14 < adiroiban1> I can add a helper to SyncTestCase ... something like skip = skipOnMissingModule('some.module')
13:14 -!- fijal [~fijal@...] has quit [Ping timeout: 250 seconds]
13:14 < adiroiban1> but not sure how often is this pattern found
13:16 < rwall> Dunno. From the description of #7014 I thought requireModule was supposed to handle this case too.
13:18 -!- fijal_ is now known as fijal
13:20 < adiroiban1> rwall true...but it ended up as a production code api. see example usage here
                    https://twistedmatrix.com/trac/attachment/ticket/6873/6873-adi-2.diff
13:21 < adiroiban1> the test skip logic is reversed to that of production code ... so I think that it needs a different api
13:21 < adiroiban1> 7014 was created based on https://twistedmatrix.com/trac/ticket/7002 ... and there it is also suggested to use namedAny
13:21 < adiroiban1> so I went for using namedAny in tests and requireModule in production
13:22 < rwall> adiroiban1: That's another question I had. Why namedAny rather than namedModule?
13:23 < adiroiban1> good question :) ... i think is inertia. namedAny is already used in other places so I prefer that method where possible
13:27 < rwall> Ok. Good enough for me I suppose. I'll check the buildbot results and merge.
13:29 < rwall> adiroiban1: Ah. https://buildbot.twistedmatrix.com/builders/py-without-modules/builds/2352/steps/trial/logs/problems
13:30 < adiroiban1> yep.... I think i should go for requireModule ...as namedAny has a strange way of raising errors
13:31 < adiroiban1> hm... but those tests did pass on my environment... i guess I don't have openssl enabled
13:35 < rwall> Well it's when OpenSSL isn't installed that we see that build failure, right?
13:35 < rwall> Or have I misunderstood.
13:35 < rwall> Anyway, I'll add some comments to the ticket.
```

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban removed owner|
|-|-|

1. fixed.


```
$ ./bin/trial --without-module OpenSSL twisted.mail.test.test_options

===============================================================================
[SKIPPED]
Missing OpenSSL package.

twisted.mail.test.test_options.OptionsTestCase.test_pop3sBackwardCompatibility
-------------------------------------------------------------------------------
Ran 13 tests in 0.045s

PASSED (skips=1, successes=12)

```


2. Not sure how to explain this.. but here is some code... see how cred.crendetial is updated by importing twisted.cred.credentials ... I don't like to rely on this kind of side effects.

```

Python 2.7.3 (default, Feb 27 2014, 19:58:35) 
[GCC 4.6.3] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from twisted import cred
>>> cred
<module 'twisted.cred' from 'twisted/cred/__init__.pyc'>
>>> cred.credentials
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'module' object has no attribute 'credentials'

>>> import twisted.cred.credentials
>>> cred.credentials
<module 'twisted.cred.credentials' from 'twisted/cred/credentials.pyc'>
>>> 

```


-------
I will attach a diff based on latest branch.

**Comment by twisted-trac** [comment]
|[<img alt="wallrj's avatar" src="https://avatars.githubusercontent.com/u/978965?s=50" width="50" height="50">](https://github.com/wallrj)<a name="note_9"></a>|@wallrj commented|
|-|-|

(In [42205]) Apply 6999-3.diff from adiroban. Refs [#6999](https://github.com/twisted/twisted/issues/6999)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @exarkun|
|-|-|

Thanks.

  1. The `twisted/mail/protocols.py` change seems wrong.  It makes the module depend on external code to import the `credentials` and `error` modules in that package - which certainly happens in the test suite since the test suite imports just about everything, but may not happen in a real application.
  1. Same for `twisted/mail/test/test_imap.py` I guess
  1. The `open(infile)` in `twisted/mail/test/test_imap.py` is kind of smelly.  `FilePath(infile).touch()` would be nicer I think.  The `open(..)` form leaks the open file object.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set status to `closed`|
|-|-|

(In [42376]) Merge pyflakes-mail-6999 

Author: adiroiban
Reviewer: rwall, exarkun
Fixes: [#6999](https://github.com/twisted/twisted/issues/6999)

Fix assorted implementation issues which pyflakes points out as unnecessary or
incorrect.
